src/aliceVision/mvsUtils/MultiViewParams.cpp

```diff
@@ -511,21 +511,36 @@ double MultiViewParams::getCamsMinPixelSize(const Point3d& x0, StaticVector<int>
     return minPixSize;
 }
 
-bool MultiViewParams::isPixelInImage(const Pixel& pix, int d, int camId) const
+bool MultiViewParams::isPixelInSourceImage(const Pixel& pixRC, int camId, int margin) const
 {
-    return ((pix.x >= d) && (pix.x < getWidth(camId) - d) && (pix.y >= d) && (pix.y < getHeight(camId) - d));
+    const IndexT viewId = getViewId(camId);
+    const sfmData::View& view = *(_sfmData.getViews().at(viewId));
+    const camera::IntrinsicBase* intrinsicPtr = _sfmData.getIntrinsicPtr(view.getIntrinsicId());
+
+    Vec2 pix_disto = intrinsicPtr->get_d_pixel({pixRC.x, pixRC.y});
+    return isPixelInImage(Pixel(pix_disto.x(), pix_disto.y()), camId, margin);
```

Comment:
All the MVS structures have a global downscale (backed in all matrices, etc.) but it's not the case for the sfmData which contains the original SfM.
So we need to compensate for that:
```
const double s = getDownscaleFactor(camId);
const Vec2 pix_disto = intrinsicPtr->get_d_pixel({pixRC.x * s, pixRC.y * s}) / s;
```